nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/util/HelmPathUtils.java

```diff
@@ -28,16 +39,59 @@
 @Named
 @Singleton
 public class HelmPathUtils
+    extends ComponentSupport
 {
+  private final IndexYamlAbsoluteUrlRewriter urlRewriter;
+
+  @Inject
+  public HelmPathUtils(final IndexYamlAbsoluteUrlRewriter urlRewriter) {
+    this.urlRewriter = checkNotNull(urlRewriter);
+  }
+
   /**
    * Returns the filename from a {@link TokenMatcher.State}.
    */
   public String filename(final TokenMatcher.State state) {
     return match(state, "filename");
   }
 
-  public String contentFilePath(final TokenMatcher.State state) {
-    return String.format("/%s",filename(state));
+  public String contentFilePath(
+      final State state,
+      final boolean isForwardingSlash)
+  {
+    String filename = filename(state);
+    return isForwardingSlash ? String.format("/%s", filename) : filename;
+  }
+
+  @Nullable
+  public String contentFileUrl(
+      final State state,
+      final Content indexYaml,
+      final boolean isForwardingSlash)
+  {
+    String filename = filename(state);
+    String chartName = getChartName(filename);
+    String chartVersion = getChartVersion(filename);
+    Optional<String> urlOpt = urlRewriter.getUrls(indexYaml, chartName, chartVersion).stream().findFirst();
+    if (urlOpt.isPresent()) {
+      String url = urlOpt.get();
+      URI uri = URI.create(url);
+      String relativeUrl = isForwardingSlash ? String.format("/%s", url) : url;
+      return uri.isAbsolute() ? url : relativeUrl;
+    }
```

Comment:
1, 2. it's not always false, please check HelmProxyFacet and HostedHandlers
3. Yes, good idea
4. Will do
5. I think class wich rewrite URLs in index.yaml obviously can have a method for URL fetching from index.yaml, isn't it? If I drop this method as private here it will be worse. Or do you want to see one more util class?
6. Yes, make sense. Just was thinking that we need more general logic here. Also will rename to getFirstUrl

Reply:
@mlukaretkyi took a look at the another method, updated, but still not sure about this method moving https://github.com/sonatype-nexus-community/nexus-repository-helm/pull/166/commits/07640cefa87b2994a992dd0b4f1e73b907c282fc